server/src/services/metiers.service.ts

```diff
@@ -1,248 +1,190 @@
   // nominal case
-  const [romesMetiers, romesDiplomes] = await Promise.all([getLabelsAndRomes(title, withRomeLabels), getLabelsAndRomesForDiplomas(title)])
+  const [romesMetiers, romesDiplomes] = await Promise.all([getMetiers({ title, withRomeLabels }), getLabelsAndRomesForDiplomas(title)])
   return { ...romesMetiers, ...romesDiplomes }
 }
 
-const getMultiMatchTerm = (term) => {
-  return {
-    bool: {
-      must: {
-        multi_match: {
-          query: term,
-          fields: [
-            "sous_domaine^80",
-            "appellations_romes^15",
-            "intitules_romes^7",
-            "intitules_rncps^7",
-            "mots_clefs_specifiques^40",
-            "domaine^3",
-            "mots_clefs^3",
-            "sous_domaine_onisep^1",
-            "intitules_fap^1",
-          ],
-          type: "phrase_prefix",
-          operator: "or",
-        },
-      },
-    },
+const searchableWeightedFields = [
+  { field: "sous_domaine_sans_accent_computed", score: 80 },
+  { field: "domaine_sans_accent_computed", score: 3 },
+  { field: "intitules_romes_sans_accent_computed", score: 7 },
+  { field: "intitules_rncps_sans_accent_computed", score: 7 },
+  { field: "mots_clefs_sans_accent_computed", score: 3 },
+  { field: "mots_clefs_specifiques_sans_accent_computed", score: 40 },
+  { field: "appellations_romes_sans_accent_computed", score: 15 },
+  { field: "intitules_fap_sans_accent_computed", score: 1 },
+  { field: "sous_domaine_onisep_sans_accent_computed", score: 1 },
+]
+
+const filterMetiers = async (regexes: any[], romes?: string, rncps?: string): Promise<(IDomainesMetiers & { score?: number })[]> => {
+  if (cacheMetiers.length === 0) {
+    await initializeCacheMetiers()
   }
-}
 
-const getMultiMatchTermForDiploma = (term) => {
-  return {
-    bool: {
-      must: {
-        multi_match: {
-          query: term,
-          fields: ["intitule_long^1", "acronymes_intitule^2"],
-          type: "phrase_prefix",
-          operator: "or",
-        },
-      },
-    },
-  }
-}
+  const results: (IDomainesMetiers & { score?: number })[] = []
 
-/**
- * retourne une liste de métiers avec leurs codes romes et codes RNCPs associés. le retour respecte strictement les critères
- */
-export const getMetiers = async ({ title, romes, rncps }: { title: string; romes?: string; rncps?: string }): Promise<{ labelsAndRomes: Omit<IMetierEnrichi, "romeTitles">[] }> => {
-  if (!title && !romes && !rncps) {
-    throw Boom.badRequest("Parameters must include at least one from 'title', 'romes' and 'rncps'")
-  } else {
-    try {
-      const terms: any[] = []
-
-      if (title) {
-        title.split(" ").forEach((term, idx) => {
-          if (idx === 0 || term.length > 2) {
-            terms.push(getMultiMatchTerm(term))
-          }
-        })
+  cacheMetiers.map((metier) => {
```

Comment:
```suggestion
  cacheMetiers.forEach((metier) => {
```
si rien n'est renvoyé par la fonction.
Le bloc peut aussi être remplacé par un `flatMap` et simplifier le code

Reply:
done